### PR TITLE
Update collaborator name when user name is updated

### PR DESCRIPTION
**Pull request by andrewmile**
This PR updates the name of any collaborator records attached to the user when the user record is updated as long as the collaborator `github_username` is the same as the user `github_username`.

This PR addresses #8 

Prior to this PR, a user can change their name on Github and when they next login to novapackages, their user name will be updated in the database but their collaborator record displays their old name on the app.

**Changed Github name from Andrew Morgan to Drew Morgan**
image

## Before
**The collaborator name did not change**
image

## After 
image

### CODE REVIEW

**Comment by mattstauffer** [comment]
@andrewmile Before I review, do we need to make any considerations for the idea that people can change their GitHub username without it changing their GitHub ID? Or is that more than we can reasonably handle?

**Comment by andrewmile** [comment]
@mattstauffer I believe this is already taken care of as shown in the test below.  If someone's `github_username` has changed we fall back to their email for authentication and then update the `github_username` on their user record.

This PR doesn't handle changing the `github_username` in the `collaborators` table because a user can have multiple collaborator records with different github usernames and we would have no way of knowing which one to update.

This PR just handles updating the `name` in the collaborators table since we can automate that easily enough.  That said, a user can manually edit their collaborator records at any time so if they have multiple collaborators with different github accounts and names they have the ability to edit them if they change their github username.

https://github.com/tightenco/novapackages/blob/master/tests/Feature/LoginTest.php#L67-L96

**Comment by mattstauffer** [comment]
OK. So if a GitHub username is changed, the *collaborator* record is never changed in our system?

I'll merge this in but can you take the charge of auditing how collaborators are or aren't modified when github usernames change, eve if that means us needing to re-work the collaborators table to track github user IDs adn stuff so that we can get updates?

**Comment by andrewmile** [comment]
Just to make sure we're on the same page, I've been referring to the following:
Github username: mattstauffer
Github name: Matt Stauffer

Is that your understanding as well?  This PR would update the collaborator `name` but never touch the collaborator `github_username`.  We could add a `github_email` address column to the `collaborators` table and use that to update the `github_username` when it changes.  Would that make sense to do or would you rather me audit how it is being used first?

**Comment by mattstauffer** [comment]
Correct--I think this made me realize we need to, separately, consider take responsibility for updating GItHub username. I'm nervous that even email address wouldn't be enough, and we may need to instead have a concept of GitHub user ID... but I'm realizing that we need to handle taht entirely separately.

File: app/User.php [hunk 1]
```diff
@@ -46,6 +46,14 @@ class User extends Authenticatable
         'password', 'remember_token',
     ];
 
+    public static function boot()
+    {
+        parent::boot();
+        self::updated(function ($user) {
```
Comment (mattstauffer) [inline app/User.php]: How hard would it be to do this only if the name is updated?

Comment (andrewmile) [reply app/User.php]: Not hard at all, just updated this :)

**Comment by andrewmile** [comment]
Yeah you're right, reliably changing the github username would require something more concrete to be in the collaborator record.  I don't remember offhand whether passport gives us the github user ID but if so, that feels a lot safer to me than the relying on the email.  

Want me to work on this as part of this PR or would you rather have it separate?

**Comment by mattstauffer** [comment]
Separate please! :) is this ready for review?

**Comment by andrewmile** [comment]
Yep!  This one is ready as is 👍

File: app/User.php [hunk 2]
```diff
@@ -46,6 +46,16 @@ class User extends Authenticatable
...
+            if ($user->isDirty('name')) {
```
Comment (mattstauffer) [inline app/User.php]: 👌

**Pull request merged**